app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,242 @@
+package agency.five.codebase.android.movieapp.ui.moviedetails
+
+import agency.five.codebase.android.movieapp.mock.MoviesMock
+import agency.five.codebase.android.movieapp.ui.component.*
+import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapper
+import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapperImpl
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.grid.*
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+
+private val movieDetailsMapper: MovieDetailsMapper = MovieDetailsMapperImpl()
+
+val movieDetailsViewState = movieDetailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails())
+
+@Composable
+fun MovieDetailsRoute(
+    onClickFavoriteButton: () -> Unit
+) {
+    val movieDetailsViewState by remember { mutableStateOf(movieDetailsViewState) }
+    MovieDetailsScreen(
+        modifier = Modifier,
+        movieDetailsViewState = movieDetailsViewState,
+        onClickFavoriteButton = onClickFavoriteButton
+    )
+}
+
+@Composable
+private fun MovieDetailsScreen(
+    modifier: Modifier,
+    movieDetailsViewState: MovieDetailsViewState,
+    onClickFavoriteButton: () -> Unit
+) {
+    Column {
+        MovieImage(
+            modifier = Modifier,
+            movieDetailsViewState = movieDetailsViewState,
+            onClickFavoriteButton = onClickFavoriteButton
+        )
+        Overview(
+            modifier = Modifier
+                .padding(bottom = 10.dp),
+            movieDetailsViewState = movieDetailsViewState
+        )
+        Crew(
+            modifier = Modifier
+                .padding(start = 10.dp, bottom = 10.dp),
+            movieDetailsViewState = movieDetailsViewState
+        )
+        Cast(
+            modifier = Modifier
+                .padding(bottom = 10.dp),
+            movieDetailsViewState = movieDetailsViewState
+        )
```

Comment:
You are passing `movieDetailsViewState` to all those functions, it would be better if functions would specify what exactly they need, and then you pass only that info. For example, `MovieImage` only needs `imageUrl`, `title`, `voteAverage` and `isFavorite`. It would be better if those were the parameters of the function.